packages/integration-tests/src/tests/api-cloud/tenant.test.ts

```diff
@@ -41,12 +49,53 @@ describe('Tenant APIs', () => {
       tag: TenantTag.Staging,
     });
     const tenant2 = await createTenant(accessToken);
+    const tenant3 = await createTenant(accessToken);
     expect(tenant1).toHaveProperty('id');
     expect(tenant1).toHaveProperty('tag', TenantTag.Staging);
     expect(tenant1).toHaveProperty('name', 'tenant1');
     expect(tenant2).toHaveProperty('id');
     expect(tenant2).toHaveProperty('tag');
     expect(tenant2).toHaveProperty('name');
+    expect(tenant3).toHaveProperty('id');
+    expect(tenant3).toHaveProperty('tag');
+    expect(tenant3).toHaveProperty('name');
+    await deleteTenant(accessToken, tenant3.id);
+    const resources = await authedAdminTenantApi.get('resources').json<Resource[]>();
+    expect(
+      resources.filter(
+        (resource) =>
+          resource.tenantId === adminTenantId &&
+          resource.indicator === getManagementApiResourceIndicator(tenant3.id)
+      ).length
+    ).toBe(0);
+    const roles = await authedAdminTenantApi.get('roles').json<Role[]>();
+    expect(
+      roles.filter(
+        (role) =>
+          role.tenantId === adminTenantId && role.name === getManagementApiAdminName(tenant3.id)
+      ).length
+    ).toBe(0);
+    const applications = await authedAdminTenantApi.get('applications').json<Application[]>();
+    expect(
+      applications.filter(
+        (application) =>
+          application.tenantId === adminTenantId &&
+          application.type === ApplicationType.MachineToMachine &&
+          application.customClientMetadata.tenantId === tenant3.id
+      ).length
+    ).toBe(0);
+    const adminConsoleApplication = applications.find(
+      (application) =>
+        application.tenantId === adminTenantId && application.id === adminConsoleApplicationId
+    );
+    expect(adminConsoleApplication).toBeDefined();
+    const urls = new GlobalValues().cloudUrlSet
+      .deduplicated()
+      .map((endpoint) => appendPath(endpoint, tenant3.id, 'callback'))
+      .map(String);
+    expect(
+      urls.every((url) => !adminConsoleApplication!.oidcClientMetadata.redirectUris.includes(url))
+    ).toBeTruthy();
```

Comment:
Sorry did not get this part. Is the couldUrlSet affected by tenants' records?

Reply:
For each tenant, we will add the corresponding redirect URI to the admin console applications configuration. The redirect URIs' rely on cloud endpoint (which can be get from cloudUrlSet).